lib/config/defaults.js

```diff
@@ -582,11 +585,37 @@ const applyOptimizationDefaults = (
  * @param {boolean} options.cache is cache enable
  * @param {boolean} options.mjs is mjs enabled
  * @param {boolean} options.webTarget is a web-like target
+ * @param {Target} options.target target
+ * @param {Mode} options.mode mode
  * @returns {void}
  */
-const applyResolveDefaults = (resolve, { cache, mjs, webTarget }) => {
+const applyResolveDefaults = (
+	resolve,
+	{ cache, mjs, webTarget, target, mode }
+) => {
 	D(resolve, "cache", cache);
 	D(resolve, "modules", ["node_modules"]);
+	F(resolve, "conditionNames", () => {
+		/** @type {string[]} */
+		const conditions = ["webpack"];
+
+		if (mode === "development") conditions.push("development");
+		if (webTarget) {
+			conditions.push("browser");
+
+			if (target === "webworker") conditions.push("worker");
+
+			return conditions;
+		}
+
+		conditions.push("node");
```

Comment:
Check if it's a node target. Users can write custom targets these should not be considered as node target by default.